File: backend/src/main/java/com/campussync/campussync_backend/dto/CreateStudentRequest.java
package com.campussync.campussync_backend.dto;

import java.time.LocalDate;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class CreateStudentRequest {

    @NotBlank
    private String name;

    @NotBlank
    @Email
    private String email;

    @NotBlank
    private String phoneNumber;

    @NotBlank
    private String registerNumber;

    @NotNull
    private LocalDate dateOfBirth;

    @NotNull
    private Long departmentId;

    @NotNull
    private Long branchId;

    @NotNull
    private Integer semester;
    @NotBlank
private String programme;

@NotNull
private Integer admissionYear;

    @NotNull
    private Integer graduationYear;
}

File: frontend/lib/models/create_student_request.dart
class CreateStudentRequest {
  final String name;
  final String email;
  final String phoneNumber;
  final String registerNumber;
  final DateTime dateOfBirth;
  final int departmentId;
  final int branchId;
  final int semester;
  final String programme;
  final int admissionYear;
  final int graduationYear;

  const CreateStudentRequest({
    required this.name,
    required this.email,
    required this.phoneNumber,
    required this.registerNumber,
    required this.dateOfBirth,
    required this.departmentId,
    required this.branchId,
    required this.semester,
    required this.programme,
    required this.admissionYear,
    required this.graduationYear,
  });

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'email': email,
      'phoneNumber': phoneNumber,
      'registerNumber': registerNumber,
      'dateOfBirth':
          dateOfBirth.toIso8601String().split('T').first,
      'departmentId': departmentId,
      'branchId': branchId,
      'semester': semester,
      'programme': programme,
      'admissionYear': admissionYear,
      'graduationYear': graduationYear,
    };
  }
}